commons/src/main/java/org/infinispan/commons/util/CollectionFactory.java

```diff
@@ -226,14 +231,38 @@
       if (requiresEquivalent(entryEq))
          return new EquivalentHashSet<T>(entryEq);
       else
-         return new HashSet<T>();
+         //return new HashSet<T>();
+         return new MiniSet<>();
    }
 
    public static <T> Set<T> makeSet(int initialCapacity, Equivalence<? super T> entryEq) {
       if (requiresEquivalent(entryEq))
          return new EquivalentHashSet<T>(initialCapacity, entryEq);
       else
-         return new HashSet<T>(initialCapacity);
+         //return new HashSet<T>(initialCapacity);
+      return new MiniSet<>();
+   }
+
+   public static <T> Set<T> makeSet(int initialCapacity) {
```

Comment:
Memory-wise, MiniSet is always at least as good as HashSet. But generally I agree, as long as the guess the caller does is correct.